provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/contract/ContractApiHttp.java

```diff
@@ -76,7 +80,10 @@
     private static final String HEADER_FIELD_VALUE_CONTENT_TYPE = "application/json";
     private static final String HEADER_FIELD_VALUE_CHARSET = "utf-8";
 
-    private final ProvisioningConnectionString provisioningConnectionString;
+    private ProvisioningConnectionString provisioningConnectionString;
+    private TokenCredentialCache credentialCache;
```

Comment:
I'd recommend copying this ```TokenCredentialCache``` class into this package rather than add a dependency on the IoT Hub service client package just for this one class.

Reply:
For context, the IoT Hub service client package has a few large dependencies of its own that the provisioning service client package doesn't need, such as the AMQP library we use.